.github/workflows/mlir-tensorrt-ci.yml

```diff
@@ -0,0 +1,227 @@
+    #paths: ['mlir-tensorrt/**']
+
+env:
+  DEFAULT_IMAGE: ghcr.io/nvidia/tensorrt-incubator/mlir-tensorrt:cuda12.5-ubuntu-llvm17
+  ADHOC_TEST_IMAGE: adhoc-test-image:cuda12.5-ubuntu-llvm17
+  REGISTRY: ghcr.io
+
+jobs:
+  mlir-tensorrt-tests:
+    if: github.event.pull_request.draft == false
+    runs-on: tripy-self-hosted
+
+    steps:
+      # Free some disk space, otherwise we get OOM error.
+      - name: Free disk space
+        run: |
+          sudo rm -rf \
+            /usr/share/dotnet "$AGENT_TOOLSDIRECTORY" /usr/local/lib/android /opt/ghc \
+            /usr/local/share/powershell /usr/share/swift /usr/local/.ghcup \
+            /usr/lib/jvm
+
+          sudo apt-get purge microsoft-edge-stable || true
+          sudo apt-get purge google-cloud-cli || true
+          sudo apt-get purge dotnet-sdk-* || true
+          sudo apt-get purge google-chrome-stable || true
+
+          sudo apt-get autoremove -y
+          sudo apt-get autoclean -y
+
+      - uses: actions/checkout@v4
+
+      # Check if we need to build a new docker image.
+      # If changes are made in Dockerfile OR python
+      # dependencies, we build a new image.
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            need_new_container:
+              - 'mlir-tensorrt/build_tools/docker/Dockerfile'
+              - 'mlir-tensorrt/python/requirements.txt'
+              - 'mlir-tensorrt/python/requirements-dev.txt'
+
+      - if: steps.filter.outputs.need_new_container == 'true'
+        run: echo "selected_test_image=${{ env.ADHOC_TEST_IMAGE }}" >> "$GITHUB_ENV"
+      - if: steps.filter.outputs.need_new_container != 'true'
+        run: echo "selected_test_image=${{ env.DEFAULT_IMAGE }}" >> "$GITHUB_ENV"
+
+      - name: Build new container, if needed
+        if: steps.filter.outputs.need_new_container == 'true'
+        uses: docker/build-push-action@v6
+        with:
+          context: mlir-tensorrt/
+          file: mlir-tensorrt/build_tools/docker/Dockerfile
+          build-args: |
+            BASE_IMAGE=nvcr.io/nvidia/cuda:12.5.1-cudnn-devel-ubuntu22.04
+            LINUX_DISTRO=ubuntu
+          tags: ${{ env.ADHOC_TEST_IMAGE }}
+          push: false
+
+      # Run initial format check
+      - name: Run python format and clang check
+        uses: addnab/docker-run-action@v3
+        with:
+          image: ${{ env.selected_test_image }}
+          options: -v ${{ github.workspace }}:/tensorrt-incubator
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          # This step does two things
+          # 1. Check if Python files follow black format
+          # 2. Check if C++ files follow clang format
+          run: |
+            cd tensorrt-incubator
+            git config --global --add safe.directory /tensorrt-incubator
+            cat > run_clang_style_check.py <<EOF
+            #!/usr/bin/env python3
```

Comment:
I believe this entire script is no longer necessary. I think you can just run the command `git clang-format HEAD~1 --diff`

Reply:
Yes. This should do the job. Thanks for pointing out